src/nlp.jl

```diff
@@ -1756,7 +1774,43 @@ function _UserFunctionEvaluator(
     return _UserFunctionEvaluator(g, ∇f, dimension)
 end
 
-# TODO: Add a docstring.
+"""
+    register(
+        model::Model,
+        s::Symbol,
+        dimension::Integer,
+        f::Function;
+        autodiff:Bool = false,
+    )
+
+Register the user-defined function `f` that takes `dimension` arguments in
+`model` as the symbol `s`.
+
+## Notes
+
+ * For this method, you must explicitly set `autodiff = true`.
```

Comment:
Specify that this is because `∇f` is not given. Can't this work without `autodiff = true` if the solver is zero order ?

Reply:
I could in theory, but it doesn't in practice. We also don't have any zero-order solvers connected. There's no real reason to use JuMP with a zero-order solver.